Makefile

```diff
@@ -56,40 +56,96 @@ TYPE :=
 # toolchain.
 LINKFLAGS ?=
 
+BUILD_TYPE := development
 ifeq ($(TYPE),)
-override LINKFLAGS += -X github.com/cockroachdb/cockroach/pkg/build.typ=development
+else ifeq ($(TYPE),msan)
+NATIVE_SUFFIX := _msan
+override GOFLAGS += -msan
+# NB: using jemalloc with msan causes segfaults. Why?
```

Comment:
I think we'd need the minimal repro to file it; for now I was discouraged by https://github.com/jemalloc/jemalloc/issues/558#issuecomment-272982746 from producing that repro, but I may revisit.

Interestingly, it doesn't look like anyone has tried using the two together, and even firefox's msan support turns off jemalloc (search for `https://developer.mozilla.org/en-US/docs/Memory_Sanitizer` in  https://developer.mozilla.org/en-US/docs/Memory_Sanitizer).